### PR TITLE
[PM-32024] Access Intelligence - CipherHealthService

### PR DESCRIPTION
**Pull request by Banrion**
## 🎟️ Tracking

[PM-32024 Access Intelligence - CipherHealthService](https://bitwarden.atlassian.net/browse/PM-32024)

### Summary

Introduces **CipherHealthService**, the first domain service in the Access Intelligence architecture rewrite. Provides unified password health analysis with concurrency control, password reuse detection, and improved performance over the existing `PasswordHealthService`.

### Files Added

- **cipher-health.view.ts**
    - View model storing password health analysis results
    - Smart model methods: `isAtRisk()`, `getPasswordStrengthLabel()`, `getPasswordStrengthBadgeVariant()`
    - Follows "Smart Models, Thin Services" pattern (domain service computes raw data, view model provides UI helpers)
- **cipher-health.service.ts**
    - Abstract class defining API contract for password health analysis
    - Platform-agnostic (no Angular dependencies - works in CLI and web)
    - Methods: `checkCipherHealth()`, `checkSingleCipherHealth()`, `detectPasswordReuse()`
- **default-cipher-health.service.ts** 
    - Reimplements password health analysis using the same underlying services (AuditService, PasswordStrengthServiceAbstraction) with architectural improvements
    - **Key improvements:**
        - Concurrency-limited HIBP calls (max 5 concurrent) prevents rate limiting for large orgs
        - Built-in password reuse detection across all ciphers
        - Returns `Map<cipherId, CipherHealthView>` for O(1) lookups (vs O(n) array searching)
        - Unified result model (vs scattered `WeakPasswordDetail`/`ExposedPasswordDetail` types)
- **default-cipher-health.service.spec.ts** 
    - Comprehensive test suite: **16 tests, all passing**
    - Coverage: All password strength scores (0-4), edge cases, concurrency control, helper methods
- **cipher-health-integration.md**
    - Integration guide for migrating from `PasswordHealthService`
    - Documents architectural changes (OLD: service computes UI labels → NEW: model provides helper methods)
    - DI setup, usage patterns, migration checklist, rollback plan

### Key Architectural Changes

**OLD:** `PasswordHealthService` mixed domain logic with UI presentation (service computed labels/badges)

**NEW:** Domain service computes raw data (scores, booleans), view model provides UI helper methods

**Benefits:**

- Separation of concerns (business logic vs presentation)
- Easier testing (service doesn't need UI knowledge)
- Reusable across platforms (CLI, web)

### CODE REVIEW

File: bitwarden_license/bit-common/src/dirt/reports/risk-insights/services/abstractions/cipher-health.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  abstract checkSingleCipherHealth(cipher: CipherView): Observable<CipherHealthView>;
+
+  /**
+   * Detects password reuse across all ciphers.
+   *
+   * Groups ciphers by password to identify reuse. Only returns passwords
+   * used by 2+ ciphers (unique passwords are excluded).
+   *
+   * @param ciphers - Array of ciphers to analyze
+   * @returns Map of password hash to array of cipher IDs sharing that password
```
Comment (ttalty) [inline bitwarden_license/bit-common/src/dirt/reports/risk-insights/services/abstractions/cipher-health.service.ts]: :pick: This says password hash yet the actual password is used from the CipherView.
